packages/@react-stately/color/src/useColorSliderState.ts

```diff
@@ -51,19 +51,35 @@ export function useColorSliderState(props: ColorSliderStateOptions): ColorSlider
       setColor(color.withChannelValue(channel, v));
     },
     onChangeEnd([v]) {
-      // onChange will have already been called with the right value, this is just to trigger onChangEnd
+      // onChange will have already been called with the right value, this is just to trigger onChangeEnd
       if (props.onChangeEnd) {
         props.onChangeEnd(color.withChannelValue(channel, v));
       }
     }
   });
 
+  function incrementThumb(index: number, stepSize: number = 1) {
+    let {maxValue, minValue, step} = color.getChannelRange(channel);
+    let channelValue = color.getChannelValue(channel);
+    let s = Math.max(stepSize, step);
+    sliderState.setThumbValue(index, channelValue + s > maxValue ? maxValue : snapValueToStep(channelValue + s, minValue, maxValue, s));
+  }
+
+  function decrementThumb(index: number, stepSize: number = 1) {
+    let {maxValue, minValue, step} = color.getChannelRange(channel);
+    let channelValue = color.getChannelValue(channel);
+    let s = Math.max(stepSize, step);
+    sliderState.setThumbValue(index, snapValueToStep(channelValue - s, minValue, maxValue, s));
+  }
```

Comment:
Do we actually need these increment/decrementThumb functions? Think we don't need them now that ColorSliderState exposes the correct step/pageStep size?

Reply:
Did some testing, I _think_ we don't need these, I believe `useSliderState`'s `increment/decrementThumb` + `useSliderThumb` is now getting the correct step/pageStep size with the changes to ColorSliderState. I'll go ahead and push the changes, note I needed to modify https://github.com/adobe/react-spectrum/blob/03d5ebea27cca652f0fa26f0b4ce369a1a65c7d0/packages/%40react-spectrum/color/test/ColorSlider.test.tsx#L304-L306, those should be expecting 1 rather than 0 since the step is 17 for PageUp/Down

Reply:
Ah wait, nevermind, we want this snapping behavior onPageUp/Down

Reply:
if a step is provided, we snap to the step, is that what you're asking?

Reply:
What about if a pageSize is provided? Should pageUp/Down always snap to a pageSize value or should it just increment/decrement by the pageSize value? Right now in this PR, ColorSlider will snap to a pageStep value upon using pageUp/Down (see the example I listed about) but this diverges from the behavior on main where pageUp/Down will increment by the pageSize value.

If it shouldn't snap to the pageStep value then I'll go ahead and remove the increment/decrementThumb functions but if that behavior is correct, then should we update the ColorArea to behave the same way

Reply:
It should snap to the step size if a step is provided, it should never snap to pageSizes unless that pageSize is a multiple of step and you're already on a step value